.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimeParser.cs

```diff
@@ -333,39 +333,49 @@ private DateTimeResolutionResult ParseSpecialTimeOfDate(string text, DateObject
         {
             var ret = new DateTimeResolutionResult();
 
-            // Handle 'eod', 'end of day'
-            var eod = this.config.EndOfDayRegex.Match(text);
-            if (eod.Success)
+            ret = ParseUnspecificTimeOfDate(text, refDateTime);
+            if (ret.Success)
             {
-                ret.Timex = DateTimeFormatUtil.FormatDate(refDateTime) + "T23:59";
-                ret.FutureValue = refDateTime.Date.AddDays(1).AddTicks(-1);
-                ret.PastValue = refDateTime.Date.AddDays(1).AddTicks(-1);
-                ret.Success = true;
-
                 return ret;
             }
-
+           
             var ers = this.config.DateExtractor.Extract(text, refDateTime);
             if (ers.Count != 1)
             {
                 return ret;
             }
 
             var beforeStr = text.Substring(0, ers[0].Start ?? 0);
-            if (this.config.TheEndOfRegex.IsMatch(beforeStr))
+            if (this.config.SpecificEndOfRegex.IsMatch(beforeStr))
             {
                 var pr = this.config.DateParser.Parse(ers[0], refDateTime);
                 var futureDate = (DateObject)((DateTimeResolutionResult)pr.Value).FutureValue;
                 var pastDate = (DateObject)((DateTimeResolutionResult)pr.Value).PastValue;
-                ret.Timex = pr.TimexStr + "T23:59";
-                ret.FutureValue = futureDate.Date.AddDays(1).AddTicks(-1);
-                ret.PastValue = pastDate.Date.AddDays(1).AddTicks(-1);
+                ret.Timex = pr.TimexStr + "T23:59:59";
```

Comment:
This logic is still duplicate here and in the ParseUnspecificTimeOfDate method. Timex generation should be centralized and future/past resolution should leverage the same method to avoid inconsistencies in the future.